cmd/create/userrole/cmd.go

```diff
@@ -0,0 +1,264 @@
+			Question: "Role prefix",
+			Help:     cmd.Flags().Lookup("prefix").Usage,
+			Default:  prefix,
+			Required: true,
+			Validators: []interactive.Validator{
+				interactive.RegExp(`[\w+=,.@-]+`),
+				interactive.MaxLength(32),
+			},
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid role prefix: %s", err)
+			os.Exit(1)
+		}
+	}
+	if len(prefix) > 32 {
+		reporter.Errorf("Expected a prefix with no more than 32 characters")
+		os.Exit(1)
+	}
+	if !aws.RoleNameRE.MatchString(prefix) {
+		reporter.Errorf("Expected a valid role prefix matching %s", aws.RoleNameRE.String())
+		os.Exit(1)
+	}
+
+	if interactive.Enabled() {
+		mode, err = interactive.GetOption(interactive.Input{
+			Question: "Role creation mode",
+			Help:     cmd.Flags().Lookup("mode").Usage,
+			Default:  aws.ModeAuto,
+			Options:  aws.Modes,
+			Required: true,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid role creation mode: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	// Get current OCM account:
+	currentAccount, err := ocmClient.GetCurrentAccount()
+	if err != nil {
+		reporter.Errorf("Failed to get current account: %s", err)
+		os.Exit(1)
+	}
+
+	switch mode {
+	case aws.ModeAuto:
+		reporter.Infof("Creating ocm user role using '%s'", creator.ARN)
+		roleARN, err := createRoles(reporter, awsClient, prefix, currentAccount.Username(), env, currentAccount.ID())
+		if err != nil {
+			reporter.Errorf("There was an error creating the ocm user role: %s", err)
+			ocmClient.LogEvent("ROSACreateUserRoleModeAuto", map[string]string{
+				ocm.Response: ocm.Failure,
+			})
+			os.Exit(1)
+		}
+		ocmClient.LogEvent("ROSACreateUserRoleModeAuto", map[string]string{
+			ocm.Response: ocm.Success,
+		})
+
+		err = userrole.Cmd.RunE(cmd, []string{roleARN})
+		if err != nil {
+			reporter.Errorf("Unable to link role arn '%s' with the account id : '%s' : %v",
+				roleARN, currentAccount.ID(), err)
+		}
+	case aws.ModeManual:
+		ocmClient.LogEvent("ROSACreateUserRoleModeManual", map[string]string{})
+		err = aws.GenerateUserRolePolicyFiles(reporter, env, currentAccount.ID())
+		if err != nil {
+			reporter.Errorf("There was an error generating the policy files: %s", err)
+			os.Exit(1)
+		}
+		if reporter.IsTerminal() {
+			reporter.Infof("All policy files saved to the current directory")
+			reporter.Infof("Run the following commands to create the account roles and policies:\n")
+		}
+		commands := buildCommands(prefix, currentAccount.Username(), creator.AccountID)
+
+		fmt.Println(commands)
+
+	default:
+		reporter.Errorf("Invalid mode. Allowed values are %s", aws.Modes)
+		os.Exit(1)
+	}
+}
+
+func buildCommands(prefix string, userName string, accountID string) string {
+	commands := []string{}
+	roleName := aws.GetUserRoleName(prefix, aws.OCMUserRole, userName)
+
+	roleARN := fmt.Sprintf("arn:aws:iam::%s:role/%s", accountID, roleName)
+	iamTags := fmt.Sprintf(
+		"Key=%s,Value=%s Key=%s,Value=%s",
+		tags.RolePrefix, prefix,
+		tags.RoleType, aws.OCMUserRole,
```

Comment:
Since the role is environment-dependent, we should add the env (integration/staging/production) as a tag